### PR TITLE
S3utils refactor

### PR DESCRIPTION
**Pull request by juandspy**
# Description

Refactored and included a new function `DeleteObjects` that will be used in Parquet Factory.

## Type of change

- New feature (non-breaking change which adds functionality)
- Refactor (refactoring code, removing useless files)

## Testing steps

Unit tests.

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: s3/deleter.go [hunk 1]
```diff
@@ -29,3 +29,22 @@ func DeleteObject(client s3iface.S3API, bucket, file string) error {
 	_, err := client.DeleteObject(input)
 	return err
 }
+
+// DeleteObjects delete the keys in the specified bucket.
+func DeleteObjects(client s3iface.S3API, bucket string, files []string) error {
+	input := &s3.DeleteObjectsInput{
+		Bucket: aws.String(bucket),
+		Delete: &s3.Delete{
```
Comment (epapbak) [inline s3/deleter.go:37]: how does s3.Delete behave if the input slice is empty?

**Review by tisnik — APPROVED** [review]
ok

File: s3/deleter.go (hunk 1 above)
Comment (epapbak) [reply s3/deleter.go:37]: btw, that could be an additional test for deleter_test.go

**Review by epapbak — APPROVED** [review]
looks good, +1

File: s3/deleter.go (hunk 1 above)
Comment (juandspy) [reply s3/deleter.go:37]: it doesn't delete anything nor raise an error: https://github.com/RedHatInsights/insights-operator-utils/pull/242/commits/320c11d3e54865b1ce253a695397e2864457f2e9

**Pull request merged**